### DISCUSSION TITLE
rpc create message fail when value exceed 1000 Fil

### DISCUSSION THREAD

**Discussion by ghost** [filecoin-project/lotus#4721] [Storage Provider] [answered]
**Describe the bug**
    
when i use the rpc to create message, if the value exceed 1000000000000000000000(1000 Fil),the message always fail,but if the value is 100000000000000000000(100 Fil),the message will succeed.

**To Reproduce**

1. Run 'use the method createMessage or estimateMessageGas'
2. See error

    2020-11-04T09:42:50.639+0800    ERROR   rpc     go-jsonrpc@v0.1.2-0.20201008195726-68c6a2704e49/server.go:87    RPC Error: unmarshaling params for 'Filecoin.GasEstimateMessageGas' (param: **types.Message): failed to parse big string: '"1e+21"'

    2020-11-04T09:42:50.639+0800    WARN    rpc     go-jsonrpc@v0.1.2-0.20201008195726-68c6a2704e49/server.go:93    rpc error: unmarshaling params for 'Filecoin.GasEstimateMessageGas' (param: **types.Message): failed to parse big string: '"1e+21"'

**Screenshots**

the code:
```javascript
const message = await wallet.createMessage({
    To: 't1wapnh3hxbl3c2cay3mefbv5p4oaljwa7wpd3rny',
    From: 't1lmifqygqvutwt65xdxmlyhwafdcb7i22nfeojja',
    Value: new Decimal('1000000000000000000000'),
});
```
local error:

UnhandledPromiseRejectionWarning: Error: Internal Server Error

node error:

2020-11-04T09:42:50.639+0800    ERROR   rpc     go-jsonrpc@v0.1.2-0.20201008195726-68c6a2704e49/server.go:87    RPC Error: unmarshaling params for 'Filecoin.GasEstimateMessageGas' (param: **types.Message): failed to parse big string: '"1e+21"'

2020-11-04T09:42:50.639+0800    WARN    rpc     go-jsonrpc@v0.1.2-0.20201008195726-68c6a2704e49/server.go:93    rpc error: unmarshaling params for 'Filecoin.GasEstimateMessageGas' (param: **types.Message): failed to parse big string: '"1e+21"'

**Version (run `lotus version`):

Daemon:  1.1.0+git.bfdb74bf+api0.17.0

Local: lotus version 1.1.0+git.bfdb74bf

go version go1.14.7 linux/amd64

**Additional context**

**Comment by jennijuju** [comment]
hrm, below works for me so  id say the issue is not in lotus rpc layer

```bash
Jennifer@jennijuju-3.local:~/filecoin/lotus curl -X POST        -H 'Content-Type: application/json'         -H "Authorization: Bearer $(cat ~/.lotus/token)"        --data '{"jsonrpc":"2.0","method":"Filecoin.MpoolPushMessage","params":[{"To":"t14pu4ogatxvok5727bd7ikp52mnnpxxxqicwyx7i","From":"t3w5afgpwisxkryfb676h7xyku57uupigdh3zdoxj3kyep53hmo5njykfatqx5rga75ra5oucbyczr7lbjaezq","Value":"1000000000000000000001","Method":0,"Params":null,"GasLimit":20000000,"GasFeeCap":"0","GasPremium":"0","Nonce":0}, {"MaxFee": "0"}],"id":0}'    'http://127.0.0.1:1234/rpc/v0'
{"jsonrpc":"2.0","result":{"Message":{"Version":0,"To":"t14pu4ogatxvok5727bd7ikp52mnnpxxxqicwyx7i","From":"t3w5afgpwisxkryfb676h7xyku57uupigdh3zdoxj3kyep53hmo5njykfatqx5rga75ra5oucbyczr7lbjaezq","Nonce":3,"Value":"1000000000000000000001","GasLimit":20000000,"GasFeeCap":"100689","GasPremium":"99635","Method":0,"Params":null,"CID":{"/":"bafy2bzacebmm2mpn65tzjp7waryokc245l2mgwj5ikdv2f27bm6nw7vy76zhu"}},"Signature":{"Type":2,"Data":"poiHiHQb0FAdxEyHd8Nhw9cqA/ifVsvlBAxtaheXcWDlStHnDx/3ITxMzHM6i/pgAkXZbztSPSY9j/ck4TSzcXz5HffRVwssNaUhdNKETNtaaAJ4m3X5BAJ8kotbbrM7"},"CID":{"/":"bafy2bzacebmm2mpn65tzjp7waryokc245l2mgwj5ikdv2f27bm6nw7vy76zhu"}},"id":0}
```

**Comment by jennijuju** [comment]
so does Filecoin.GasEstimateMessageGas

Jennifer@jennijuju-3.local:~/filecoin/lotus curl -X POST        -H 'Content-Type: application/json'         -H "Authorization: Bearer $(cat ~/.lotus/token)"        --data '{"jsonrpc":"2.0","method":"Filecoin.GasEstimateMessageGas","params":[{"To":"t14pu4ogatxvok5727bd7ikp52mnnpxxxqicwyx7i","From":"t3w5afgpwisxkryfb676h7xyku57uupigdh3zdoxj3kyep53hmo5njykfatqx5rga75ra5oucbyczr7lbjaezq","Value":"1000000000000000000001","Method":0,"Params":null,"GasLimit":20000000,"GasFeeCap":"0","GasPremium":"0","Nonce":0}, {"MaxFee": "0"},null],"id":0}'    'http://127.0.0.1:1234/rpc/v0'
{"jsonrpc":"2.0","result":{"Version":0,"To":"t14pu4ogatxvok5727bd7ikp52mnnpxxxqicwyx7i","From":"t3w5afgpwisxkryfb676h7xyku57uupigdh3zdoxj3kyep53hmo5njykfatqx5rga75ra5oucbyczr7lbjaezq","Nonce":0,"Value":"1000000000000000000001","GasLimit":20000000,"GasFeeCap":"100823","GasPremium":"99769","Method":0,"Params":null,"CID":{"/":"bafy2bzacecdubaecf57wmymxhn7symgos7ne772fluv5u4amxw7zo5lcqs7f2"}},"id":0}

**Comment by jennijuju** [comment, marked as answer]
so id suggest you recheck your wallet library implementation, which one are you using?

Converting this to discussion so that you can get some help from our ecosystem!

**Reply by ghost** [reply]
thank you,I find the value can not use the Decimal or BigNumber,if I use the string '1000000000000000000000'.it will succeed
